pkg/artifact/job.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifact
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/abcxyz/github-metrics-aggregator/pkg/version"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/workerpool"
+)
+
+// ExecuteJob runs the ingestion pipeline job to read GitHub
+// action workflow logs from GitHub and store them into GCS.
+func ExecuteJob(ctx context.Context, cfg *Config) error {
+	logger := logging.FromContext(ctx)
+
+	bq, err := NewBigQuery(ctx, cfg.ProjectID, cfg.DatasetID)
+	if err != nil {
+		return fmt.Errorf("failed to create bigquery client: %w", err)
+	}
+	defer bq.Close()
+
+	eventsTableDotNotation := formatGoogleSQL(cfg.ProjectID, cfg.DatasetID, cfg.EventsTableID)
+	artifactsTableDotNotation := formatGoogleSQL(cfg.ProjectID, cfg.DatasetID, cfg.ArtifactsTableID)
+
+	// Create a pool of workers to manage all of the log ingestions
+	pool := workerpool.New[ArtifactRecord](&workerpool.Config{
+		Concurrency: int64(runtime.NumCPU()),
+		StopOnError: false,
+	})
+
+	// Setup a log ingester to process ingestion events
+	logsFn, err := NewLogIngester(ctx, cfg.BucketName, cfg.GitHubAppID, cfg.GitHubInstallID, cfg.GitHubPrivateKey)
+	if err != nil {
+		return fmt.Errorf("failed to create log ingester: %w", err)
+	}
+
+	logger.InfoContext(ctx, "ingestion job starting",
+		"name", version.Name,
+		"commit", version.Commit,
+		"version", version.Version)
+
+	// Read up to `BatchSize` number of events that need to be processed
+	query := fmt.Sprintf(SourceQuery, eventsTableDotNotation, artifactsTableDotNotation, cfg.BatchSize)
```

Comment:
Similar to @verbanicm's comment about SQLi, it doesn't seem like `eventsTableDotNotation` or `artifactsTableDotNotation` are used anywhere besides here. I wonder if we should create a function off of `cfg` instead that returns the string. That way the "sql compiling logic" is all in a single place, and we could inject sanitization there in the future if these values became user input.